ios/RCCNavigationController.m

```diff
@@ -272,10 +274,27 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
     NSString *title = button[@"title"];
     UIImage *iconImage = nil;
     id icon = button[@"icon"];
+    NSString *customView = button[@"customView"];
     if (icon) iconImage = [RCTConvert UIImage:icon];
-    
+    NSString *buttonId = button[@"id"];
+    NSString *callbackId = button[@"onPress"];
+
     UIBarButtonItem *barButtonItem;
-    if (iconImage)
+    if (customView) {
+      RCTBridge *bridge = ((RCTRootView*)viewController.view).bridge;
+
+      NSNumber *viewWidth = button[@"customViewWidth"] ?: @60;
+      NSMutableDictionary *initialProps = [NSMutableDictionary dictionaryWithDictionary:button[@"customViewInitialProps"]];
+      initialProps[@"buttonID"] = buttonId;
+      initialProps[@"callbackID"] = callbackId;
+      RCTRootView *reactView = [[RCTRootView alloc] initWithBridge:bridge moduleName:customView initialProperties:initialProps];
+      CGRect buttonFrame = CGRectMake(0, 0, viewWidth.floatValue, self.navigationBar.bounds.size.height);
```

Comment:
Coming with iOS 11, navigation bars will change height depending on the `prefersLargeTitles`, meaning that we'd have to also listen for changes there and act accordingly as well. I'd prefer to see views have some arbitrary height as defined by the React component's styling and have Yoga size it and feed out a view of that size, then allow for iOS to place it in the navigation bar as needed. This will allow for consumers to create components of arbitrary heights rather than forcing one on them.